include/Server/src/Server.cpp

```diff
@@ -93,84 +93,103 @@ void SmtpServer::Start()
     Logger::LogDebug("Exiting Start");
 }
 
-void SmtpServer::Accept() {
+void SmtpServer::Accept()
+{
     Logger::LogDebug("Entering SmtpServer::Accept");
     auto new_socket = std::make_shared<TcpSocket>(m_io_context);
 
     Logger::LogProd("Ready to accept new connections.");
 
-    m_acceptor->async_accept(*new_socket, [this, new_socket](const boost::system::error_code& error) {
-        if (!error) {
-            Logger::LogProd("Accepted new connection.");
-            m_thread_pool.EnqueueDetach([this, new_socket]() { HandleClient(SocketWrapper(new_socket)); });
-        } else {
-            ErrorHandler::HandleBoostError("Accept", error);
-        }
-        Accept();
-    });
+    m_acceptor->async_accept(
+        *new_socket,
+        [this, new_socket](const boost::system::error_code& error)
+        {
+            if (!error)
+            {
+                Logger::LogProd("Accepted new connection.");
+                m_thread_pool.EnqueueDetach([this, new_socket]() { HandleClient(SocketWrapper(new_socket)); });
+            }
```

Comment:
I think, you may have troubles with database during multithreading. If I am right, each thread works with single `CommandHandler` which has single instance of `PgMailDB` (PgMailDB establishes only one connection with database), but this leads to problem because one connection to database could have only one transaction (do one work at time). For each thread you need to create its own connection with database. 

Possible solution:

- Create new connection to database like you create new socket. Rewrite methods for `CommandHandler` so that they accept PgMailDB or even IMailDB by pointer or reference. Rewrite other methods if needed.

**Correct me, if am not right.**